core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -133,7 +133,11 @@ class SearchService(
         val check = FileCheck(base)
         val indexed = extractSymbolsFromClassOrJar(base).flatMap(persist(check, _, commitIndex = false, boost = boost))
         indexed.onComplete { _ => semaphore.release() }
-        indexed
+        indexed.recover {
+          case t: Throwable =>
+            log.warn(s"Failed to index ${base.getName()}.", t)
```

Comment:
I think I'd need to see a convincing example of when this fails to understand why it needs to be handled here instead of caught earlier on. I don't like adding in arbitrary recovery code when we can catch the problems before doing any work.